modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,425 @@
+		$html_input     = "<input type='checkbox' $checked_text value='1'
+			name='attachments[{$post->ID}][show_dominant_color]' id='attachments[{$post->ID}][show_dominant_color]'/>
+			<span style='background-color: #{$dominant_color}; color: {$contrast_color}; padding: 3px 6px; margin-top: 3px;vertical-align: middle;display: inline-block; '>#{$dominant_color}</span>";
+
+		$form_fields['show_dominant_color'] = array(
+			'label' => __( 'Show Dominant color as a preload placeholder', 'wp-dominant-color' ),
+			'input' => 'html',
+			'html'  => $html_input,
+		);
+
+		return $form_fields;
+	}
+
+	/**
+	 * Save dominant color setting value for a media post.
+	 *
+	 * @param array $post
+	 * @param array $attachment
+	 *
+	 * @return array
+	 */
+	public function save_dominant_color_media_setting( array $post, array $attachment ) {
+		$attachment_id = $post['ID'];
+		$image_meta    = wp_get_attachment_metadata( $attachment_id );
+
+		if ( isset( $attachment['show_dominant_color'] ) ) {
+			if ( ! isset( $image_meta['show_dominant_color'] ) ) {
+				// If enabling blurhash from media setting and image meta doesn't have any, generate a new one.
+				$image_meta                        = $this->dominant_color_metadata( $image_meta, $attachment_id );
+				$image_meta['show_dominant_color'] = $attachment['show_dominant_color'] ? '1' : '0';
+				wp_update_attachment_metadata( $attachment_id, $image_meta );
+			}
+		} elseif ( isset( $image_meta['dominant_color'] ) ) {
+			// If disabling blurhash from media setting and blurhash set in image meta, unset it.
+			unset( $image_meta['show_dominant_color'] );
+			wp_update_attachment_metadata( $attachment_id, $image_meta );
+		}
+
+		return $post;
+	}
+
+	/**
+	 * @param $id
+	 * @param $default_color
+	 *
+	 * @return string
+	 */
+	public function get_dominant_color( $id, $default_color = 'eee' ) {
+		$file         = get_attached_file( $id );
+		$image_editor = _wp_image_editor_choose( $file );
+		if ( $image_editor === 'WP_Image_Editor_GD' ) {
+			return $this->get_dominant_color_GD( $file, $default_color );
+		} elseif ( $image_editor === 'WP_Image_Editor_Imagick' ) {
+			return $this->get_dominant_color_Imagick( $file, $default_color );
+		}
+
+		return $default_color;
+	}
+
+	/**
+	 * @param $file
+	 * @param $default_color
+	 *
+	 * @return string
+	 */
+	private function get_dominant_color_GD( $file, $default_color = 'eee' ) {
+
+		wp_raise_memory_limit( 'image' );
+		if ( $original_image = @imagecreatefromstring( @file_get_contents( $file ) ) ) {
+			$shortend_image = imagecreatetruecolor( 1, 1 );
+			imagecopyresampled( $shortend_image, $original_image, 0, 0, 0, 0, 1, 1, imagesx( $original_image ), imagesy( $original_image ) );
+
+			return dechex( imagecolorat( $shortend_image, 0, 0 ) );
+		} else {
+			return $default_color;
+		}
+	}
+
+	/**
+	 * @param $file
+	 * @param $default_color
+	 *
+	 * @return string
+	 */
+	private function get_dominant_color_Imagick( $file, $default_color = 'eee' ) {
```

Comment:
I am not sure about the use of default color, this might have side effects we do not understand yet.